auto-farm/src/farms_whitelist.rs

```diff
@@ -0,0 +1,85 @@
+use crate::address_to_id_mapper::{AddressId, AddressToIdMapper, NULL_ID};
+
+elrond_wasm::imports!();
+
+#[elrond_wasm::module]
+pub trait FarmsWhitelistModule:
+    crate::farm_external_storage_read::FarmExternalStorageReadModule + utils::UtilsModule
+{
+    #[only_owner]
+    #[endpoint(addFarms)]
+    fn add_farms(&self, farms: MultiValueEncoded<ManagedAddress>) {
+        let farms_mapper = self.farm_ids();
+        for farm_addr in farms {
+            self.require_sc_address(&farm_addr);
+
+            let new_id = farms_mapper.get_id_or_insert(&farm_addr);
+            if new_id == NULL_ID {
+                continue;
+            }
+
+            let farm_config = self.get_farm_config(&farm_addr);
+            self.farm_for_farm_token(&farm_config.farm_token_id)
+                .set(new_id);
```

Comment:
If `new_id` already exists, wouldn't `set` fail?
`get_id_or_insert` returns an existing id -> set; else -> update

Reply:
Also for `.insert` below.

Reply:
Fixed, it would always add farm addresses even if they were existing already.